.github/workflows/main-build.yml

```diff
@@ -0,0 +1,64 @@
+name: Build
+
+on:
+  pull_request:
+    branches:
+      - main
+
+env:
+  CARGO_TERM_COLOR: always
+  ARCH_TYPE: sm_70
```

Comment:
probably minor - assuming it's a local runner,  I would set it to sm_86 or sm_89 depending on where it sits